### PR TITLE
Unit tests for casper sending blocks

### PR DESCRIPTION
**Pull request by birchmd**
## Overview
-Instance of `TransportLayer` for packet-sending testing purposes
-Factor out test instances for effects into reusable (node-like) class
-Factory method for multiple instances of this class, all of which can communicate
-Unit tests for Casper using new class

### Does this PR relate to an RChain JIRA issue? 
https://rchain.atlassian.net/browse/RHOL-379

### Complete this checklist before you submit the PR
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### Notes
Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else.

### CODE REVIEW

**Comment by KentShikama** [comment]
I think @rabbitonweb should take another look at the transport layer part before merging though.

File: casper/src/test/scala/coop/rchain/casper/util/comm/TransportLayerTestImpl.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package coop.rchain.casper.util.comm
```
Comment (rabbitonweb) [inline casper/src/test/scala/coop/rchain/casper/util/comm/TransportLayerTestImpl.scala]: We have a stub like that in `comm`, maybe worth reusing existing one (extending it if needed)? Does not have to be in this PR but don't forget about it. The smaller number of stubs the better.

File: comm/src/main/scala/coop/rchain/p2p/effects/PacketHandler.scala [hunk 2]
```diff
@@ -19,6 +19,21 @@ object PacketHandler extends PacketHandlerInstances {
         C.handlePacket(peer, packet).liftM[T]
     }
 
+  def pf[F[_]: Applicative: Log](
```
Comment (rabbitonweb) [inline comm/src/main/scala/coop/rchain/p2p/effects/PacketHandler.scala]: cool, nice to see it moved

File: casper/src/test/scala/coop/rchain/casper/util/comm/TransportLayerTestImpl.scala (hunk 1 above)
Comment (birchmd) [reply casper/src/test/scala/coop/rchain/casper/util/comm/TransportLayerTestImpl.scala]: I looked at using it, but it seemed to be specialized for testing `roundTrip` as opposed to `send` and `receive`, which Casper uses. I didn't see an easy way to extend it for my purposes, which is why I wrote my own. I did use the `comm` stubs for all the other effects though.

**Pull request merged**